spec/requests/posts_spec.rb

```diff
@@ -1,17 +1,42 @@
 require 'rails_helper'
 
-RSpec.describe 'Posts', type: :request do
-  describe 'GET /index' do
-    it 'returns http success' do
-      get '/posts/index'
-      expect(response).to have_http_status(:success)
+RSpec.describe PostsController, type: :controller do
+  let!(:user) { User.create(name: 'Ben', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.') }
+  let!(:post) do
+    user.posts.create(title: 'Test Post', text: 'This is a test post.', comments_counter: 0, likes_counter: 0)
+  end
+
+  describe 'GET #index' do
+    it 'returns a success response' do
+      get :index, params: { user_id: user.id }
+      expect(response).to be_successful
+    end
+
+    it 'renders the index template' do
+      get :index, params: { user_id: user.id }
+      expect(response).to render_template('index')
+    end
+
+    it 'assigns @posts' do
+      get :index, params: { user_id: user.id }
+      expect(assigns(:posts)).to eq([post])
     end
   end
 
-  describe 'GET /show' do
-    it 'returns http success' do
-      get '/posts/show'
-      expect(response).to have_http_status(:success)
+  describe 'GET #show' do
+    it 'returns a success response' do
+      get :show, params: { user_id: user.id, id: post.id }
+      expect(response).to be_successful
```

Comment:
- [ ] [REQUIRED] Good job so far! Please make sure that for `posts show` you add tests that check 🙏🏼 

- If the response status was correct.

- If the response body includes correct placeholder text.